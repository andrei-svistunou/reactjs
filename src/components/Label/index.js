import React from 'react';
import PropTypes from 'prop-types';

const Label = props => (
  <span className="label">{props.text}</span>
);

Label.propTypes = {
  text: PropTypes.string.isRequired
};

export default Label;
