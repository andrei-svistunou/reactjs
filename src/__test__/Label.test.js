import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Label from '../components/Label';

describe('<Label />', () => {
  test('should render <Label /> with props', () => {
    const logo = shallow(<Label text='test label' />);

    expect(logo.text()).toEqual('test label');
    expect(toJson(logo)).toMatchSnapshot();
  });
});
