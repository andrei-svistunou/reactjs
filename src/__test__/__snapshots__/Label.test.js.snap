// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> should render <Label /> with props 1`] = `
<span
  className="label"
>
  test label
</span>
`;
